src/vtkF3DAlembicImporter.cxx

```diff
@@ -0,0 +1,265 @@
+#include <vtkImageData.h>
+#include <vtkImageReader2.h>
+#include <vtkImageReader2Factory.h>
+#include <vtkLight.h>
+#include <vtkMatrix4x4.h>
+#include <vtkObjectFactory.h>
+#include <vtkPointData.h>
+#include <vtkPoints.h>
+#include <vtkPolyData.h>
+#include <vtkPolyDataMapper.h>
+#include <vtkProperty.h>
+#include <vtkQuaternion.h>
+#include <vtkRenderer.h>
+#include <vtkShaderProperty.h>
+#include <vtkSmartPointer.h>
+#include <vtkStringArray.h>
+#include <vtkTexture.h>
+#include <vtkTransform.h>
+#include <vtkUniforms.h>
+#include <vtkUnsignedShortArray.h>
+
+#include <vtksys/SystemTools.hxx>
+
+#if defined(_MSC_VER)
+#pragma warning(push, 0)
+#endif
+#include <Alembic/AbcGeom/All.h>
+#include <Alembic/AbcCoreOgawa/All.h>
+#include <Alembic/Util/All.h>
+#include <Alembic/Abc/All.h>
+#include <Alembic/AbcCoreAbstract/All.h>
+#include <Alembic/AbcCoreFactory/All.h>
+#if defined(_MSC_VER)
+#pragma warning(pop)
+#endif
+
+vtkStandardNewMacro(vtkF3DAlembicImporter);
+
+
+class vtkF3DAlembicImporterInternalEx
+{
+	typedef std::map<Alembic::AbcGeom::PlainOldDataType,std::string> PODStringMap;
+	typedef std::map<Alembic::AbcGeom::PropertyType,std::string> PropertyTypeStringMap;
+public:
+	vtkF3DAlembicImporterInternalEx(vtkF3DAlembicImporter* parent) {
+		this->Parent = parent;
+	}
+	void CreatePODStringMap(PODStringMap& podStringMap)
+	{
+		podStringMap[Alembic::AbcGeom::kBooleanPOD] = "kBooleanPOD";
+		podStringMap[Alembic::AbcGeom::kUint8POD] = "kUint8POD";
+		podStringMap[Alembic::AbcGeom::kInt8POD] = "kInt8POD";
+		podStringMap[Alembic::AbcGeom::kUint16POD] = "kUint16POD";
+		podStringMap[Alembic::AbcGeom::kInt16POD] = "kInt16POD";
+		podStringMap[Alembic::AbcGeom::kUint32POD] = "kUint32POD";
+		podStringMap[Alembic::AbcGeom::kInt32POD] = "kInt32POD";
+		podStringMap[Alembic::AbcGeom::kUint64POD] = "kUint64POD";
+		podStringMap[Alembic::AbcGeom::kInt64POD] = "kInt64POD";
+		podStringMap[Alembic::AbcGeom::kFloat16POD] = "kFloat16POD";
+		podStringMap[Alembic::AbcGeom::kFloat32POD] = "kFloat32POD";
+		podStringMap[Alembic::AbcGeom::kFloat64POD] = "kFloat64POD";
+		podStringMap[Alembic::AbcGeom::kStringPOD] = "kStringPOD";
+		podStringMap[Alembic::AbcGeom::kWstringPOD] = "kWstringPOD";
+		podStringMap[Alembic::AbcGeom::kNumPlainOldDataTypes] = "kNumPlainOldDataTypes";
+		podStringMap[Alembic::AbcGeom::kUnknownPOD] = "kUnknownPOD";
+	}
+	void CreatePropertyTypeStringMap(PropertyTypeStringMap& propertyTypeStringMap)
+	{
+		propertyTypeStringMap[Alembic::AbcGeom::kCompoundProperty] = "kCompoundProperty";
+		propertyTypeStringMap[Alembic::AbcGeom::kScalarProperty] = "kScalarProperty";
+		propertyTypeStringMap[Alembic::AbcGeom::kArrayProperty] = "kArrayProperty";
+	}
+	void ProcessIPolyMesh(vtkRenderer* renderer, const Alembic::AbcGeom::IPolyMesh& pmesh)
+	{
+		vtkNew<vtkPoints> points;
+		vtkNew<vtkCellArray> polys;
+		vtkNew<vtkPolyData> polydata;
+
+		Alembic::AbcGeom::IPolyMeshSchema::Sample samp;
+		if ( pmesh.getSchema().getNumSamples() > 0 )
```

Comment:
should be `if (condition)`
there is a clang format file at the root of F3D, you can use it to automatically format your files.

Reply:
Done